### PR TITLE
Adding rubygem-smart_proxy_discovery_image

### PR DESCRIPTION
**Pull request by lzap**
Initial version.

### CODE REVIEW

**Comment by lzap** [comment]
Submitting build with: koji -c ~/.koji/katello-config build --nowait --scratch foreman-plugins-nightly-nonscl-rhel6 /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.3-1.git.0.8bac993.el6.src.rpm
Uploading srpm: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.3-1.git.0.8bac993.el6.src.rpm
[                                    ]  00% 00:00:00     0.00 B        - B/sec
[====================================] 100% 00:00:00  21.37 KiB  29.57 KiB/sec
Created task: 339232
Task info: http://koji.katello.org/koji/taskinfo?taskID=339232

Wrote: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.3-1.git.0.8bac993.git.0.8bac993.el7.src.rpm



Submitting build with: koji -c ~/.koji/katello-config build --nowait --scratch foreman-plugins-nightly-nonscl-rhel7 /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.3-1.git.0.8bac993.git.0.8bac993.el7.src.rpm
Uploading srpm: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.3-1.git.0.8bac993.git.0.8bac993.el7.src.rpm
[                                    ]  00% 00:00:00     0.00 B        - B/sec
[====================================] 100% 00:00:00  21.40 KiB  30.57 KiB/sec
Created task: 339233
Task info: http://koji.katello.org/koji/taskinfo?taskID=339233

Wrote: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.3-1.git.0.8bac993.git.0.8bac993.git.0.8bac993.fc19.src.rpm



Submitting build with: koji -c ~/.koji/katello-config build --nowait --scratch foreman-plugins-nightly-fedora19 /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.3-1.git.0.8bac993.git.0.8bac993.git.0.8bac993.fc19.src.rpm
Uploading srpm: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.3-1.git.0.8bac993.git.0.8bac993.git.0.8bac993.fc19.src.rpm
[                                    ]  00% 00:00:00     0.00 B        - B/sec
[====================================] 100% 00:00:00  21.41 KiB  27.70 KiB/sec
Created task: 339235
Task info: http://koji.katello.org/koji/taskinfo?taskID=339235

**Comment by lzap** [comment]
Updated to 1.0.3 due to issues in gemspec.

File: rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+mkdir -p %{buildroot}%{foreman_proxy_settingsd_dir}
+mv %{buildroot}%{gem_instdir}/settings.d/discovery_image.yml.example \
+   %{buildroot}%{foreman_proxy_settingsd_dir}/discovery_image.yml
+
+%files
+%dir %{gem_instdir}
+%{gem_libdir}
+%{gem_spec}
+%{foreman_proxy_bundlerd_dir}/discovery_image.rb
+%config(noreplace) %{foreman_proxy_settingsd_dir}/discovery_image.yml
```
Comment (domcleal) [inline rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec]: %config is probably better

File: rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+%global foreman_proxy_bundlerd_dir %{foreman_proxy_dir}/bundler.d
+%global foreman_proxy_settingsd_dir %{_sysconfdir}/foreman-proxy/settings.d
+
+Name: rubygem-%{gem_name}
+Version: 1.0.3
+Release: 1%{?dist}
+Summary: Discovery image (node) plugin for Foreman's smart proxy
+Group: Applications/Internet
+License: GPLv3
+URL: http://github.com/theforeman/smart_proxy_discovery_image
```
Comment (domcleal) [inline rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec]: This doesn't match the gemspec, one of them's wrong?

File: rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+BuildRequires: ruby(rubygems)
+
+BuildArch: noarch
+
+Provides: rubygem(%{gem_name}) = %{version}
+Provides: foreman-proxy-plugin-discovery-image
+
+%description
+This smart proxy plugin is not intended for direct use. It is installed
+on foreman discovery image which is running on discovered nodes.
```
Comment (domcleal) [inline rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec]: Capitalise Foreman Discovery?  It also doesn't match the gemspec, it's probably worth syncing them.

**Comment by domcleal** [comment]
I'm concerned that users may install this by accident as it has a similar name to the regular plugin.  As far as I can tell, the plugin doesn't use the smart proxy's authorisation features which could lead to remote execution or shutdown of servers (subject to sudo).  Any thoughts about making this more inactive by default?

File: rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+Release: 1%{?dist}
+Summary: Discovery image (node) plugin for Foreman's smart proxy
+Group: Applications/Internet
+License: GPLv3
+URL: http://github.com/theforeman/smart_proxy_discovery_image
+Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires: ruby(rubygems)
+Requires: foreman-proxy >= 1.7.0
+Requires: rubygem(rest-client) > 1.6.2
```
Comment (domcleal) [inline rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec]: This dep isn't in the gem.

File: rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec (hunk 2 above)
Comment (lzap) [inline rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec]: Fixed in the gem, thanks.

File: rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec (hunk 3 above)
Comment (lzap) [inline rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec]: Fixing.

**Comment by lzap** [comment]
Hmmm I can put a warning in the example configuration plus make it `false` the default value.

**Comment by lzap** [comment]
Looks reasonable I think:

```diff
diff --git a/settings.d/discovery_image.yml.example b/settings.d/discovery_image.yml.example
index 09143aa..fb4f20c 100644
--- a/settings.d/discovery_image.yml.example
+++ b/settings.d/discovery_image.yml.example
@@ -1,2 +1,6 @@
 ---
-:enabled: true
+# Smart Proxy plugin providing Image API on discovered nodes. This plugin is only
+# useful on discovered nodes, do not install on regular Smart Proxy deployments.
+# WARNING: If you enable this plugin on standard deployment, your host will be
+# vulnerable to remote reboot and arbitrary code execution!
+:enabled: false
```

**Comment by domcleal** [comment]
Yep, that would be great.

**Comment by lzap** [comment]
Submitting build with: koji -c ~/.koji/katello-config build --nowait --scratch foreman-plugins-nightly-nonscl-rhel6 /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.4-1.git.0.0a568a0.el6.src.rpm
Uploading srpm: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.4-1.git.0.0a568a0.el6.src.rpm
[                                    ]  00% 00:00:00     0.00 B        - B/sec
[====================================] 100% 00:00:00  21.57 KiB  26.62 KiB/sec
Created task: 340840
Task info: http://koji.katello.org/koji/taskinfo?taskID=340840

Wrote: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.4-1.git.0.0a568a0.git.0.0a568a0.el7.src.rpm



Submitting build with: koji -c ~/.koji/katello-config build --nowait --scratch foreman-plugins-nightly-nonscl-rhel7 /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.4-1.git.0.0a568a0.git.0.0a568a0.el7.src.rpm
Uploading srpm: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.4-1.git.0.0a568a0.git.0.0a568a0.el7.src.rpm
[                                    ]  00% 00:00:00     0.00 B        - B/sec
[====================================] 100% 00:00:00  21.59 KiB  29.12 KiB/sec
Created task: 340845
Task info: http://koji.katello.org/koji/taskinfo?taskID=340845

Wrote: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.4-1.git.0.0a568a0.git.0.0a568a0.git.0.0a568a0.fc19.src.rpm



Submitting build with: koji -c ~/.koji/katello-config build --nowait --scratch foreman-plugins-nightly-fedora19 /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.4-1.git.0.0a568a0.git.0.0a568a0.git.0.0a568a0.fc19.src.rpm
Uploading srpm: /tmp/tito/rubygem-smart_proxy_discovery_image-1.0.4-1.git.0.0a568a0.git.0.0a568a0.git.0.0a568a0.fc19.src.rpm
[                                    ]  00% 00:00:00     0.00 B        - B/sec
[====================================] 100% 00:00:00  21.60 KiB  28.53 KiB/sec
Created task: 340846
Task info: http://koji.katello.org/koji/taskinfo?taskID=340846

File: rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec (hunk 4 above)
Comment (domcleal) [inline rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec]: Still an issue - should it be removed?

Comment (lzap) [inline rubygem-smart_proxy_discovery_image/rubygem-smart_proxy_discovery_image.spec]: Oh missed that!

**Comment by lzap** [comment]
Pushed fixed version.

**Comment by domcleal** [comment]
Thanks @lzap, building.

**Pull request merged**